.github/actions/deploy-happy-stack/action.yml

```diff
@@ -38,18 +42,26 @@ runs:
         CREATE_TAG: ${{ inputs.create-tag }}
         TAG: ${{ inputs.tag }}
         ENV: ${{ inputs.env }}
+        OPERATION: ${{ inputs.operation }}
       shell: bash
       run: |
         set -ue
         set -o pipefail
         
         IMAGE_TAG="${TAG:-sha-$GITHUB_SHA}"
         echo "Targetting Stack ${STACK_NAME}, tagging with '${IMAGE_TAG}'"
+        HAPPY_DEPLOYMENT_FLAGS=""
 
-        if $(happy list --aws-profile "" --env=${ENV} 2>&1 | grep -q ${STACK_NAME} ); then
-          echo "Updating stack ${STACK_NAME}"
-          happy update --aws-profile "" --env=${ENV} --create-tag=${CREATE_TAG} --tag ${IMAGE_TAG} ${STACK_NAME}
-        else
-          echo "Creating stack ${STACK_NAME}"
-          happy create --aws-profile "" --env=${ENV} --create-tag=${CREATE_TAG} --tag ${IMAGE_TAG} ${STACK_NAME}
+        if [[ ${OPERATION} == "create-or-update" ]]; then
+          HAPPY_DEPLOYMENT_FLAGS="--create-tag=${CREATE_TAG} --tag ${IMAGE_TAG}"
```

Comment:
nit picking: if this is set below anyway, why not change the condition check below to if from elif, so that only one place sets the flags?

Reply:
My intent was to make the distinction between what to do if an operation input was given and if an input wasn't given. Only setting the flags in one place feels better though. Will change.